File: 8_Kyu_Kata/count_of_positives_slash_sum_of_negatives/JavaScript/countOfPositivesSlashSumOfNegatives.js
// 8 kyu - Count of positives / sum of negatives  [ ID: 576bb71bbbcf0951d5000044  (count-of-positives-slash-sum-of-negatives) ]
// URL: https://www.codewars.com/kata/576bb71bbbcf0951d5000044
// Category: undefined  |  Tags: FUNDAMENTALS | ARRAYS | LISTS
// *****************************************************************************
function countPositivesSumNegatives(input) {
    if (input && input != [0])
      if (input.filter(el => el > 0).length == 0 && input.filter(el => el < 0).reduce((a,b) => a+b,0) == 0) return []
      else return [input.filter(el => el > 0).length, input.filter(el => el < 0).reduce((a,b) => a+b,0)]
    return []
  }

File: 8_Kyu_Kata/count_of_positives_slash_sum_of_negatives/Python/count_of_positives_slash_sum_of_negatives.py
# 8 kyu - Count of positives / sum of negatives  [ ID: 576bb71bbbcf0951d5000044  (count-of-positives-slash-sum-of-negatives) ]
# URL: https://www.codewars.com/kata/576bb71bbbcf0951d5000044
# Category: NONE  |  Tags: FUNDAMENTALS | ARRAYS | LISTS
# ******************************************************************************
def count_positives_sum_negatives(arr):
      result = []
      if len(arr):    #test for empty list
          result.append(len(list(filter(lambda x: x>0, arr)))) #filter input for positives into a list and append length to result
          result.append(sum(list(filter(lambda x: x<0, arr)))) #filter input for negatives into a list and append sum to result
      return result   #return either empty list if "if" test failed or the result of the 2 "if" appends
